### PR TITLE
Null pointers should not be dereferenced

### PR DESCRIPTION
**Pull request by tsauerwein**
Replacement for https://github.com/mapfish/mapfish-print/pull/381 (rebased and style issues fixed)

### CODE REVIEW

File: core/src/main/java/org/mapfish/print/processor/http/matcher/AddressHostMatcher.java [hunk 1]
```diff
@@ -130,6 +130,9 @@ public int hashCode() {
 
     @Override
     public boolean equals(Object obj) {
+        if(obj == null) {
```
Comment (sbrunner) [inline core/src/main/java/org/mapfish/print/processor/http/matcher/AddressHostMatcher.java]: missing space

File: core/src/main/java/org/mapfish/print/processor/http/matcher/DnsHostMatcher.java [hunk 2]
```diff
@@ -115,6 +115,9 @@ public int hashCode() {
 
     @Override
     public boolean equals(Object obj) {
+        if(obj == null) {
```
Comment (sbrunner) [inline core/src/main/java/org/mapfish/print/processor/http/matcher/DnsHostMatcher.java]: idem

**Comment by sbrunner** [comment]
Looks good :-)

**Comment by tsauerwein** [comment]
Thanks for the review.

**Pull request merged**